File: go/difference_of_squares.go
package diffsquares

func SquareOfSum(n int) int {
	sum := (1 + n) * n / 2
	return sum * sum
}

func SumOfSquares(n int) (sum int) {
	sum = n * (n + 1) * (2*n + 1) / 6
	return
}

func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}

File: rust/difference_of_squares.rs
pub fn square_of_sum(n: u32) -> u32 {
    let sum: u32 = (1..=n).into_iter().sum();
    sum * sum
}

pub fn sum_of_squares(n: u32) -> u32 {
    (1..=n).into_iter().map(|x| x * x).sum()
}

pub fn difference(n: u32) -> u32 {
    square_of_sum(n) - sum_of_squares(n)
}
